db/dcrpg/tableupgrades.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2018, The dcrdata developers
+// See LICENSE for details.
+
+package dcrpg
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/decred/dcrd/rpcclient"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/dcrdata/db/dbtypes"
+	"github.com/decred/dcrdata/db/dcrpg/internal"
+	"github.com/decred/dcrdata/rpcutils"
+)
+
+// CheckForAuxDBUpgrade checks if an upgrade is required and currently supported.
+// A boolean value is returned to indicate that the db upgrade was successfully completed.
+func (pgb *ChainDB) CheckForAuxDBUpgrade(dcrdClient *rpcclient.Client) (bool, error) {
+	var (
+		version     = TableVersion{}
+		upgradeInfo = TableUpgradesRequired(TableVersions(pgb.db))
+	)
+
+	if len(upgradeInfo) > 0 {
+		version = upgradeInfo[0].RequiredVer
+	} else {
+		return false, nil
+	}
+
+	switch {
+	case upgradeInfo[0].UpgradeType != "upgrade":
+		return false, nil
+
+	// When the required table version is 2.x.0 where x is greater than or equal to 3
+	case version.major == 2 && version.minor > 2 && version.patch == 0:
```

Comment:
This needs to work for upgrades over multiple versions.   So if there is a 2.4.0, and the software is just updated from 2.0.0, it will need to run each upgrade step in sequence.  I *think* this approach can be adapted to work that way, but give it some thought too please.